node_modules/gh-users/tests/util.js

```diff
@@ -37,6 +37,89 @@ var assertGetMe = module.exports.assertGetMe = function(client, callback) {
+ * @param  {String}             [update.email]                  The updated email
+ * @param  {String}             [update.emailPreference]        The updated email preference
+ * @param  {Function}           callback                        Standard callback function
+ * @param  {User}               callback.user                   The updated user
+ * @throws {AssertionError}                                     Error thrown when an assertion failed
+ */
+var assertUpdateUser = module.exports.assertUpdateUser = function(client, id, update, callback) {
+    client.user.updateUser(id, update, function(err, user) {
+        assert.ok(!err);
+        assert.ok(user);
+        return callback(user);
+    });
+};
+
+/**
+ * Assert that a user can not be updated
+ *
+ * @param  {RestClient}         client                          The REST client to make the request with
+ * @param  {Number}             id                              The id of the user to update
+ * @param  {Object}             update                          The updates to make
+ * @param  {String}             [update.displayName]            The updated displayName
+ * @param  {String}             [update.email]                  The updated email
+ * @param  {String}             [update.emailPreference]        The updated email preference
+ * @param  {Number}             code                            The expected HTTP error code
+ * @param  {Function}           callback                        Standard callback function
+ * @throws {AssertionError}                                     Error thrown when an assertion failed
+ */
+var assertUpdateUserFails = module.exports.assertUpdateUserFails = function(client, id, update, code, callback) {
+    client.user.updateUser(id, update, function(err, user) {
+        assert.ok(err);
+        assert.strictEqual(err.code, code);
+        assert.ok(!user);
+        return callback();
+    });
+};
+
+/**
+ * Assert that the users for an application can be retrieved
+ *
+ * @param  {RestClient}         client                          The REST client to make the request with
+ * @param  {Number}             [app]                           The application to search users in, defaults to the current application
+ * @param  {String}             [filter]                        The string to filter users by
+ * @param  {Number}             [limit]                         The maximum number of results to retrieve. Defaults to 10
+ * @param  {Number}             [offset]                        The number to start paging from. Defaults to 0
+ * @param  {Function}           callback                        Standard callback function
+ * @param  {User[]}             callback.users                  The users in the application
+ * @throws {AssertionError}                                     Error thrown when an assertion failed
+ */
+var getUsers = module.exports.getUsers = function(client, app, filter, limit, offset, callback) {
```

Comment:
With all the randomisation and similarity matching that's going on, it's a bit tricky. Might be best to leave this to the individual tests